reporter/http/http.go

```diff
@@ -244,6 +247,7 @@ func NewReporter(url string, opts ...ReporterOption) reporter.Reporter {
 	}
 
 	go r.loop()
+	go r.batchLoop()
```

Comment:
I think send should directly append to the queue of traces to send. removing the need for the second goroutine.

remember that a channel is a triple of (mutex, conditional variable, array). Here the client code appends to the channel (lock, append, unlock, condnotify). Then the first goroutine pumps from the queue (condwait, lock, remove item, unlock) then appends to the queue of pending traces (lock, append, potential resize, unlock).

You can halve the work and contention!

Reply:
I'm not entirely convinced this is equivalent, so I'll make the other suggested changes first.

I think the blocking cases are equivalent; with or without the extra goroutine, `Send` can be blocked waiting for either `batchSend` or `append` to give up the `batchMtx`.

In the non-blocking case, `Send` hands over the span to the `loop` goroutine and immediately returns. Without that goroutine, `Send` would only return after the `append` function finishes. This would be a small change to userspace, but it might be larger than the benefit of eliminating a goroutine and the `spanC` channel.